extensions/amp-mustache/0.2/amp-mustache.js

```diff
@@ -58,27 +59,59 @@ export class AmpMustache extends AMP.BaseTemplate {
     /** @private @const {string} */
     this.template_ = this.initTemplateString_();
 
-    mustache.parse(this.template_, /* tags */ undefined);
+    mustache.parse(this.template_, /* tags */ this.getDelimiters_());
   }
 
   /**
    * @private
    * @return {string}
    */
   initTemplateString_() {
-    if (this.element.tagName == 'TEMPLATE') {
-      const content = templateContentClone(this.element);
+    const {element} = this;
+    if (element.tagName == 'TEMPLATE') {
+      const content = templateContentClone(element);
       this.processNestedTemplates_(content);
-      const container = this.element.ownerDocument.createElement('div');
+      const container = element.ownerDocument.createElement('div');
       container.appendChild(content);
       return container./*OK*/innerHTML;
-    } else if (this.element.tagName == 'SCRIPT') {
-      return this.element.textContent;
+    } else if (element.tagName == 'SCRIPT') {
+      return element.textContent;
     }
 
     return '';
   }
 
+  /**
+   * Initialize the delimiters.
+   * @return {?Array<string>} delimiters or null. Null rather than empty array
+   *     so that default mustache delimiters are used.
+   * @private
+   */
+  getDelimiters_() {
+    const CUSTOM_DELIMITERS_ATTR = 'data-custom-delimiters';
+    let delimiters = null;
+    const {element} = this;
+    if (element.hasAttribute(CUSTOM_DELIMITERS_ATTR)) {
+      const delimitersStr = element.getAttribute(CUSTOM_DELIMITERS_ATTR);
+      delimiters = delimitersStr.split(',');
+      userAssert(delimiters.length == 2,
+          'Beginning and ending delimiter is required: %s.', element);
+      // If using a template encode any html entities used in a delimiter.
+      // This is required as the template will also contain decoded delimiters
+      // prior to being parsed by mustache.
+      if (element.tagName == 'TEMPLATE') {
+        const textArea = document.createElement('textarea');
+        for (let i = 0; i < delimiters.length; i++) {
+          const delimiter = delimiters[i];
+          textArea.textContent = delimiter;
+          delimiters[i] = textArea./*OK*/innerHTML;
+        }
```

Comment:
Nit: It's not obvious what this code is doing here, so a well-named helper function may help readability:

```js
if (element.tagName == 'TEMPLATE') {
  delimiters = htmlEncode(delimiters);
}
```